docs/types/reference-types.rst

```diff
@@ -470,6 +470,114 @@ Array Members
 
 .. index:: ! array;slice
 
+Dangling References to Storage Array Elements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When working with storage arrays, you need to take care to avoid dangling references.
+A dangling reference can for example occur, if you store a reference to an array element in
+a local variable and then ``.pop()`` from the containing array:
```

Comment:
But we do not have storage references to value types. You can have references to stuff in storage, but only to reference types. I think it would be good to point that out because it's not necessarily obvious and we sometimes rely on this. For example you need to know this to see why the impact of changing the string from short to long is limited to this weird `push()`/tuple case.

Reply:
I added a sentence about this after the first example.

By the way: it's not only ``(byteArray.push(), byteArray.push()) = ...``, but it can also be
``(byteArray[0], byteArray.push()) = ...`` (or the other way round) - should I make that explicit or is it fine as is? The warnings in the other PR should hit both cases.